worlds/pokemon_emerald/locations.py

```diff
@@ -173,18 +185,9 @@ def set_free_fly(world: "PokemonEmeraldWorld") -> None:
     # If not enabled, set it to Littleroot Town by default
     fly_location_name = "EVENT_VISITED_LITTLEROOT_TOWN"
     if world.options.free_fly_location:
-        fly_location_name = world.random.choice([
-            "EVENT_VISITED_SLATEPORT_CITY",
-            "EVENT_VISITED_MAUVILLE_CITY",
-            "EVENT_VISITED_VERDANTURF_TOWN",
-            "EVENT_VISITED_FALLARBOR_TOWN",
-            "EVENT_VISITED_LAVARIDGE_TOWN",
-            "EVENT_VISITED_FORTREE_CITY",
-            "EVENT_VISITED_LILYCOVE_CITY",
-            "EVENT_VISITED_MOSSDEEP_CITY",
-            "EVENT_VISITED_SOOTOPOLIS_CITY",
-            "EVENT_VISITED_EVER_GRANDE_CITY",
-        ])
+        free_fly_locations = sorted(map(lambda city: BLACKLIST_OPTION_TO_VISITED_EVENT[city], world.options.free_fly_blacklist.value))
+        if free_fly_locations:
+            fly_location_name = world.random.choice(free_fly_locations)
```

Comment:
This is inverted, isn't it? For this code, `free_fly_locations` contains the event names for the cities that the user marked (blacklisted). So we want to choose from the full list minus the blacklist. That's why I had `set` operations in that comment. And the `sorted` only needs to be applied because it turns a set back into a list with deterministic order (no need to sort the `map`, in other words).

For the record, the community supported the blacklist version of this option, so we're going with that even though I suggested a whitelist earlier.

Reply:
And while I'm here, a comprehension/generator looks better than `map` to me.
```py
set(BLACKLIST_OPTION_TO_VISITED_EVENT[city] for city in world.options.free_fly_blacklist.value)
```
This creates a set of event-named strings from the player's option. And then you can subtract it from the set of all of them. And then make a sorted list of the result.